### ISSUE TITLE
Notify users of new nearby competitions

### ISSUE THREAD

**Issue by jbcm627** [cubing/worldcubeassociation.org#114]
I've had a couple questions about existence of such a thing recently.

This was a ccm idea too, but it might fit better here.
https://github.com/cubing/ccm/issues/108

**Comment by jfly** [comment]
Note: This would require asking users where they live. Would we also want different distance thresholds for different parts of the world? @pedrosino has been thinking about this stuff recently, what do you think?

**Comment by pedrosino** [comment]
I think we could use a fixed one. We'd have to figure out what is close for each country and then store it someway. Not worth the trouble, I think.

**Comment by TimMcMahon** [comment]
@jfly Could users nominate Country that they live in so that delegates in that Region can field questions/admin tasks?

**Comment by jfly** [comment]
Good idea, although it's probably worth mentioning at this point that we don't have a way to go from an iso2 country code to a list of delegates for that iso2 country code.

**Comment by EvilPotato1216** [comment]
Basically, anything like country or delegate could be used to track down competitions near the user. The user could select a specific delegate and get notified about their new competitions.

**Comment by EvilPotato1216** [comment]
I also think the user could enter their own mile/km radius around them where the competition should happen for them to get a notification. The notification should be via email I think.

**Comment by EvilGnome6** [comment]
CubingUSA used to have such a feature (although it never seemed to work). They have optional fields for City/State you can add to your profile. The feature used a selectable range (100 miles, 500 miles, etc.) to receive notifications of new competitions.

I think adding something like this would be worth the effort. I would love to go to some competitions in neighboring States where I'm not the delegate but I never hear about them until it's too late. I also get complaints from people who miss my competition registration deadlines because they didn't hear about it in time.

**Comment by EvilPotato1216** [comment]
Yeah, looking back at this more than a year later, I think this could be in the profile edit area. Users could select a km radius or a specific delegate to be notified of their competitions. Whole new avenue of features for this one.

**Comment by timreyn** [comment]
I'm going to look into this.  Some suggestions came up on the staff list related to #3797 

My proposal: a user can create one or more Subscription, with the following parameters:

- Either within X km of (lat, lng), or all competitions in a country
- May limit to competitions with a particular event
- Optionally, automatically bookmark the competition when it's announced
- Optionally, email the user when the competition is announced

So, for example, I could subscribe to all competitions within 500km of my home, and also all competitions in the US with Fewest Moves.

**Comment by lgarron** [comment]
@timreyn, that sounds pretty fab!

> My proposal: a user can create one or more Subscription, with the following parameters:

Supporting multiple would be very useful to me. :-D
I travel to Europe once or twice a year, and it would be useful to keep an eye out for adjacent competitions, especially if I can specify a date range!

As an additional step, it would also be nice to have a competition calendar. :-D #64

**Comment by saranshgrover** [comment]
@timreyn do you have plans on working/completing this?

**Comment by timreyn** [comment]
Sorry, this (and everything else cubing related) got deprioritized last year when COVID started.

I can add this back to my todo list, though there's still a decent amount more work on top of what I did last year.

**Comment by dunkOnIT** [comment]
@Jambrose777 From a data protection perspective, do you know what constraints would apply in the following cases:
1. Allowing the user to "subscribe" to competitions in a particular country
2. Allowing the user to "subscribe" to competitions in a certain radius of a set of coordinates (eg pin placed on a map)
3. Allowing the user to "subscribe" to competitions in a certain radius of a set of coordinates

**Comment by Jambrose777** [comment]
> @Jambrose777 From a data protection perspective, do you know what constraints would apply in the following cases:
> 
> 1. Allowing the user to "subscribe" to competitions in a particular country
> 2. Allowing the user to "subscribe" to competitions in a certain radius of a set of coordinates (eg pin placed on a map)
> 3. Allowing the user to "subscribe" to competitions in a certain radius of a set of coordinates

All that is fine. Users giving consent to use their data in specific ways is completely fine. We should allow them to unsubscribe at any time / remove that consent.
